packages/api/src/routers/external-api/v2/charts.ts

```diff
@@ -0,0 +1,351 @@
+import { ClickhouseClient } from '@hyperdx/common-utils/dist/clickhouse';
+import { getMetadata } from '@hyperdx/common-utils/dist/metadata';
+import {
+  ChartConfigWithOptDateRange,
+  DisplayType,
+} from '@hyperdx/common-utils/dist/types';
+import { SourceKind } from '@hyperdx/common-utils/dist/types';
+import opentelemetry, { SpanStatusCode } from '@opentelemetry/api';
+import express from 'express';
+import _ from 'lodash';
+import { z } from 'zod';
+import { validateRequest } from 'zod-express-middleware';
+
+import { Granularity } from '@/clickhouse';
+import { getConnectionById } from '@/controllers/connection';
+import { getSource } from '@/controllers/sources';
+import { getTeam } from '@/controllers/team';
+import { IConnection } from '@/models/connection';
+import { ISource } from '@/models/source';
+import { translateExternalSeriesToInternalSeries } from '@/utils/externalApi';
+import {
+  externalQueryChartSeriesSchema,
+  objectIdSchema,
+  timeChartSeriesSchema,
+} from '@/utils/zod';
+
+// Mapping from API's MetricsDataType enum (title-case) to common-utils format (lowercase)
+const normalizeMetricDataType = (metricType?: string): string | undefined => {
+  if (!metricType) return undefined;
+
+  // Convert first letter to lowercase to match common-utils expectations
+  return metricType.charAt(0).toLowerCase() + metricType.slice(1);
+};
```

Comment:
I will clean this up. 

TL;DR zod schema uses MetricDataType from the clickhouse package which has values that are title cased. Everywhere else, we use MetricDataType from common-utils which has values that are lowercase.